File: haskell/iniciante/1047.hs
-- title: Tempo de Jogo com Minutos
-- number of the problem: 1047
-- date: 22/09/20
-- author: ltsdw

main :: IO ()
main = do
    l1 <- getLine

    let (init_hours : init_minutes : final_hours : final_minutes : _) = map (read :: String -> Int) $ words l1

    let r_hours     = final_hours - init_hours
    let r_h_j       = r_hours
    let r_minutes   = final_minutes - init_minutes
    let r_m_j       = r_minutes

    let r_hours     = if r_h_j < 0 then r_h_j + 24 else r_h_j
    let r_minutes   = if r_m_j < 0 then r_m_j + 60 else r_m_j
    let r_h_j       = r_hours
    let r_hours     = if r_m_j < 0 then r_h_j - 1 else r_h_j
    let r_h_j       = r_hours
    let r_hours     = if r_h_j < 0 then r_h_j + 24 else r_h_j

    if init_hours == final_hours && init_minutes == final_minutes
        then putStrLn "O JOGO DUROU 24 HORA(S) E 0 MINUTO(S)"
    else
        putStrLn $ "O JOGO DUROU " ++ show r_hours ++ " HORA(S) E " ++ show r_minutes ++ " MINUTO(S)"

File: c/iniciante/1047.c
/* title: Tempo de Jogo com Minutos
 * problem number: 1047
 * date: 16/09/20
 * author: ltsdw
 */

#include <stdio.h>

void timeOfGame(int init_hours, int init_minutes, int final_hours, int final_minutes);

int main(void) {
    int init_hours, init_minutes, final_hours, final_minutes;
    scanf("%d%d%d%d", &init_hours, &init_minutes, &final_hours, &final_minutes);

    timeOfGame(init_hours, init_minutes, final_hours, final_minutes);

    return 0;
}

void timeOfGame(int init_hours, int init_minutes, int final_hours, int final_minutes) {
    int r_hours = final_hours - init_hours;
    int r_minutes = final_minutes - init_minutes;

    if (r_hours < 0) {
        r_hours += 24;

    }

    if (r_minutes < 0 ) {
        r_minutes += 60;
        r_hours--;
        if (r_hours < 0) {
            r_hours += 24;
        }
    }

    if ( (init_hours == final_hours) && (init_minutes == final_minutes) ) {
        printf("O JOGO DUROU 24 HORA(S) E 0 MINUTO(S)\n");
    } else {
        printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", r_hours, r_minutes);
    }
    return;
}
